src/FxCop/Desktop.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithmsAnalyzer.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+using Desktop.Analyzers.Common;
+
+
+namespace Desktop.Analyzers
+{
+    public abstract class DoNotUseInsecureCryptographicAlgorithmsAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string DoNotUseMD5RuleId = "CA5350";
```

Comment:
So … what’s next step?

From: Gen Lu [mailto:notifications@github.com]
Sent: Wednesday, October 7, 2015 8:41 AM
To: dotnet/roslyn-analyzers <roslyn-analyzers@noreply.github.com>
Cc: Rick Qing Xu <qinx@microsoft.com>
Subject: Re: [roslyn-analyzers] Add FxCop Crypto rules and Xml rules (#304)


In src/FxCop/Desktop.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithmsAnalyzer.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2froslyn-analyzers%2fpull%2f304%23discussion_r41406170&data=01%7c01%7cqinx%40microsoft.com%7c4231496d2980490b74cc08d2cf2da344%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=SE4EFYOlxnoRpF9NH0lNYpvq5tb%2fBzI3hiA%2bDD8NhR8%3d>:

> +

> +using System.Collections.Generic;

> +using System.Collections.Immutable;

> +using System.Linq;

> +

> +using Microsoft.CodeAnalysis;

> +using Microsoft.CodeAnalysis.Diagnostics;

> +

> +using Desktop.Analyzers.Common;

> +

> +

> +namespace Desktop.Analyzers

> +{

> +    public abstract class DoNotUseInsecureCryptographicAlgorithmsAnalyzer : DiagnosticAnalyzer

> +    {

> +        internal const string DoNotUseMD5RuleId = "CA5350";

@michaelcfanning<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fmichaelcfanning&data=01%7c01%7cqinx%40microsoft.com%7c4231496d2980490b74cc08d2cf2da344%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=9gcVnSdmXa3mBQaNEuVZp36vvjo0symcNfx4aIK2Wss%3d> I actually did implement the analyzer with a single ID, but then the team made a decision to keep using old IDs instead. One of the reasons, IIRC, is some algorithms is less broken than others and we want to give users the ability to choose.

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2froslyn-analyzers%2fpull%2f304%2ffiles%23r41406170&data=01%7c01%7cqinx%40microsoft.com%7c4231496d2980490b74cc08d2cf2da344%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=dNgh9%2fBiTbZ%2fmt1uHcmWRm6XZN2N%2flgNmADR0iKEwuE%3d>.